packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1210,22 +1214,9 @@ export default function buildKernel(
     // directly.
 
     if (crankResults.abort) {
-      // errors unwind any changes the vat made
-      kernelKeeper.abortCrank();
-
-      // some deliveries should be consumed when they fail
-      if (crankResults.popDelivery) {
-        // kernelKeeper.abortCrank removed all evidence that the crank
-        // ever happened, including, notably, the removal of the
-        // delivery itself from the head of the run queue, which will
-        // result in it being delivered again on the next crank. If we
-        // don't want that, then we need to remove it again.
-
-        // eslint-disable-next-line no-use-before-define
-        getNextDeliveryMessage();
-      }
-      // other deliveries should be re-attempted on the next crank, so they
-      // get the right error: we leave those on the queue
+      kernelKeeper.rollbackCrank(
```

Comment:
let's preserve a few comments:

```js
if (crankResults.abort) {
  // errors unwind any changes the vat made, by rolling back to the "deliver" savepoint
  // In addition, the crankResults will either ask for the message to be consumed
  // (without redelivery), or they'll ask for it to be attempted again (so it can go "splat"
  // against a terminated vat, and give the sender the right error). In the latter case,
  // we roll back to the "start" savepoint, established by `run()` or `step()` before
  // the delivery was pulled off the run-queue, effectively pushing it back on.
```